app/collection.go

```diff
@@ -7,37 +7,49 @@ import (
 	"net/http"
 
 	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/utils"
 )
 
 func (a *App) registerCollectionAndTopic(pluginID, collectionType, topicType string) error {
-	collectionTypeForPlugin, ok := a.ch.collectionTypes[pluginID]
-	if ok && collectionTypeForPlugin != collectionType {
+	// we have a race condition due to multiple plugins calling this method
+	a.ch.collectionMut.Lock()
+	defer a.ch.collectionMut.Unlock()
+
+	// check if collectionType was already registered by other plugin
+	if pluginIDForCollection, found := getKeyWithValue(a.ch.collectionTypes, collectionType); found && pluginIDForCollection != pluginID {
```

Comment:
nit: `getKeyWithValue` doesn't help me understand what that method actually does. Given we only use it once and it's relatively short, should we just inline for clarity?